revert: "fix: replace_plugin does not work as expected with .ts config (#5920)"

IWANABETHATGUY opened the pull request:
> Now that https://github.com/rolldown/rolldown/issues/5941 is resolved we don't have any dual publishing hazard > problems anymore. Can https://github.com/rolldown/rolldown/pull/5920 then be reverted or is it good to keep anyway?

Closed #6073

IWANABETHATGUY commented:
* **#6074** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6074?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6074?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>graphite: merge</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/rolldown?ref=merge-queue-instructions-in-stack-comment&prId=6074).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen reviewed (commented):
PR title should be descriptive.

IWANABETHATGUY commented:
### Merge activity

* **Sep 8, 3:38 PM UTC**: The merge label 'graphite: merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown) once it meets the requirements.
* **Sep 8, 3:52 PM UTC**: `IWANABETHATGUY` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown).
* **Sep 8, 4:03 PM UTC**: The [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown) couldn't merge this PR because **it was not satisfying all requirements** (Failed CI: 'node-test (macos-latest) / Node Test').
* **Sep 8, 4:49 PM UTC**: The merge label 'graphite: merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown) once it meets the requirements.
* **Sep 8, 4:49 PM UTC**: `IWANABETHATGUY` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown).
* **Sep 8, 4:49 PM UTC**: Merged by the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown).

IWANABETHATGUY committed:
revert: "fix: replace_plugin does not work as expected with .ts config (#5920)" (#6074)

> Now that https://github.com/rolldown/rolldown/issues/5941 is resolved we don't have any dual publishing hazard > problems anymore. Can https://github.com/rolldown/rolldown/pull/5920 then be reverted or is it good to keep anyway?

Closed #6073
